lib/tasks/retention_instructions.rake

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+# $ bundle exec rake "retention_instructions:backfill[batch_size]"
+
+require_relative '../../app/helpers/retention_instruction_helper'
+
+namespace :retention_instructions do
+
+  include RetentionInstructionHelper
+  desc 'Backfill retention instructions'
+
+  task :backfill, [:batch_size] => :environment do |_, args|
+    args.with_defaults(batch_size: 1000)
+    batch_size = args[:batch_size].to_i
+
+    puts "Backfilling retention instructions with batch size: #{batch_size}..."
+
+    ActiveRecord::Base.transaction do
+      saved_count = 0
+      Labware.where(retention_instruction: nil).find_each(batch_size: batch_size) do |labware|
+        saved_count = process_labware(labware, saved_count)
+      end
+      puts "Backfilled retention instructions for #{saved_count} labware items."
+    end
+
+  end
+
+  # rubocop:todo Metrics/AbcSize
+  # rubocop:todo Metrics/MethodLength
+  def process_labware(labware, saved_count)
+    return saved_count unless labware.custom_metadatum_collection.present? &&
+      labware.custom_metadatum_collection.metadata['retention_instruction'].present?
+
+    labware.retention_instruction = find_retention_instruction_key_for_value(
+      labware.custom_metadatum_collection.metadata['retention_instruction']
+    ).to_sym
+
+    begin
+      labware.custom_metadatum_collection.custom_metadata.where(key: 'retention_instruction').find_each(&:destroy!)
+      saved_count += 1 if labware.save!
+      saved_count
+    rescue ActiveRecord::ActiveRecordError => e
+      puts "Error processing labware #{labware.id}: #{e.message}"
+      raise e
+    rescue StandardError => e
+      puts "Error processing labware #{labware.id}: #{e.message}"
+    end
```

Comment:
When can a StandardError happen in this block ? I am asking because we are not rolling back the transaction in this case.

Reply:
Yeah I think it should also raise. I added the standard error to make it more generalisable. I didn't actually find a case where it explicitly throw a `StandardError` though.

Reply:
I see you have now re-throwing the exception in that block similar to the other. I am resolving this convo.